src/sc_crawler/cli.py

```diff
@@ -50,6 +57,39 @@ def metadata_dump(sql, *_args, **_kwargs):
     SQLModel.metadata.create_all(engine)
 
 
+@cli.command()
+def hash(
+    connection_string: Annotated[
+        str, typer.Option(help="Database URL with SQLAlchemy dialect.")
+    ] = "sqlite:///sc_crawler.db",
+    level: Annotated[
+        HashLevels,
+        typer.Option(
+            help="Return hashes for the whole database, per table or for each row."
+        ),
+    ] = "database",
+):
+    engine = create_engine(connection_string)
+
+    with Session(engine) as session:
+        table_hashes = {table.get_table_name(): table.hash(session) for table in tables}
+
+    if level == HashLevels.TABLE:
+        table_hashes = {
+            k: sha1(dumps(v, sort_keys=True).encode()).hexdigest()
+            for k, v in table_hashes.items()
+        }
+        print(dumps(table_hashes, sort_keys=True))
+        raise typer.Exit()
+
+    json_dump = dumps(table_hashes, sort_keys=True)
+    if level == HashLevels.ROW:
+        print(json_dump)
+
+    if level == HashLevels.DATABASE:
+        print(sha1(json_dump.encode()).hexdigest())
```

Comment:
Provide a consistent output type here maybe? Either JSON or a formatted text output (not sure about the `ROW` type, but for the others, the latter might be the best?).

Reply:
good point :+1: 

what about adding a `format` option, defaults to JSON, but supporting YAML as well. that should be good for DB and TABLE levels, and good enough for ROW as well

Reply:
> good point 👍
> 
> what about adding a `format` option, defaults to JSON, but supporting YAML as well? that should be good for DB and TABLE levels, and good enough for ROW as well

In FreeBSD a lot of tools use [libxo](https://github.com/Juniper/libxo), which provides a similar capability for generic CLI tools. Nice idea, although I'm not sure if anybody will ever use it. :)

Reply:
the only reason to have two output types in my opinion was to provide a way to return a single hash without curly braces :) I think I'll just drop the `--level` support and keep it at the DB level, and for any other levels, python helper can be used